setup

```diff
@@ -199,12 +204,20 @@ def build(args, remaining):
     # clear cmake cache
     if args.do_force_rebuild and os.path.exists(builddir): shutil.rmtree(builddir)
 
+    if args.do_compile_commands and os.path.exists(builddir):
+        logging.error("Generating compile_commands.json on a non-clean build may miss some files")
+        return -1
+
     # create directories
     if not os.path.exists(builddir): os.makedirs(builddir)
     if not os.path.exists(installdir): os.makedirs(installdir)
 
     # build up args string for the cmake command
-    cmake_cmd = "cmake " + rootdir + " -DCMAKE_INSTALL_PREFIX=" + installdir
+    cmake_cmd = ""
+    if args.do_compile_commands:
+        cmake_cmd += "bear --"
```

Comment:
Do we want to add an entry in the docs for how to run `bear` externally, which may be useful for devs? Maybe here: https://shadow.github.io/docs/guide/coding.html

Reply:
Done